csi/socketclient.py

```diff
@@ -0,0 +1,98 @@
+import os
+import json
+import socket
+import uuid
+import logging
+
+from kadalulib import execute
+
+SOCKET_FILE_PATH = "/kadalu/glusterfs/glustersocket.sock"
+GLUSTERFS_CMD = "/usr/sbin/glusterfs"
+MOUNT_CMD = "/usr/bin/mount"
+HOSTVOL_MOUNTDIR = "/mnt/kadalu"
+is_connected = False
+
+def connectSocket(client_socket):
+    retry_interval = 60
+    max_retries = 5
+
+    retry_count = 0
+    connected = False
+    while retry_count < max_retries and not connected:
+        try:
+            # Attempt to connect to the server
+            client_socket.connect(SOCKET_FILE_PATH)
+            connected = True
+            logging.info("Connected to the server!")
+            break
+        except ConnectionRefusedError:
+            # Connection refused, wait for a while before retrying
+            logging.info("Connection refused. Retrying in seconds..." + str(retry_interval))
+            time.sleep(retry_interval)
+            retry_count += 1
+
+    logging.info("the connection state is " + str(connected))
+    return connected
+
+def parseCommand(commandList):
```

Comment:
The client only runs this command in this class afaict, so this is unnecessary

Reply:
But as we discussed earlier, there are many other commands in this class that are being executed, since for this case we only need glusterfs command to be executed on vmexec, I added this check.

Reply:
Ah, ok